tensorizer/serialization.py

```diff
@@ -3526,13 +3526,15 @@ def _pwrite_syscall(
             verify if isinstance(verify, int) else self._buffer_size(data)
         )
         bytes_just_written: int = os.pwrite(self._fd, data, offset)
-        bytes_written += bytes_just_written
+        if bytes_just_written > 0:
+            bytes_written += bytes_just_written
```

Comment:
Hmm. Usually, that's `0` or greater. Under what situations does `pwrite` return a negative number?

Reply:
`pwrite` returns a negative number if it errored. This should hopefully be caught in CPython's wrapper around the actual syscall and converted to a Python exception, but just in case, it won't add it to the accumulator either.